src/main/Plugins/ValidationPlugins/Dns/DnsValidation.cs

```diff
@@ -13,35 +14,59 @@ namespace PKISharp.WACS.Plugins.ValidationPlugins
     /// </summary>
     public abstract class DnsValidation<TOptions, TPlugin> : Validation<TOptions, Dns01ChallengeValidationDetails>
     {
-        public DnsValidation(ILogService logService, TOptions options, string identifier) : 
-            base(logService, options, identifier) { }
+        protected readonly DomainParser _domainParser;
+        protected readonly ILookupClientProvider _lookupClientProvider;
+
+        public DnsValidation(DomainParser domainParser, ILookupClientProvider lookupClientProvider, ILogService logService, TOptions options, string identifier) : 
+            base(logService, options, identifier)
+        {
+            _domainParser = domainParser;
+            _lookupClientProvider = lookupClientProvider;
+        }
 
         public override void PrepareChallenge()
         {
             CreateRecord(_challenge.DnsRecordName, _challenge.DnsRecordValue);
             _log.Information("Answer should now be available at {answerUri}", _challenge.DnsRecordName);
 
-            string foundValue = null;
             try
             {
-                if (IPAddress.TryParse(Properties.Settings.Default.DnsServer, out IPAddress address)) 
+                var domainName = _domainParser.Get(_challenge.DnsRecordName);
+                var lookupClient = _lookupClientProvider.Default;
+
+                _log.Debug("Querying SOA for {Domain}", domainName.RegistrableDomain);
+                var soaResponse = lookupClient.Query(domainName.RegistrableDomain, QueryType.SOA);
+
+                var nameServer = soaResponse.Answers.SoaRecords().FirstOrDefault()?.MName;
```

Comment:
Since we're going into the deep end, doesn't it make sense to actually check all of the name servers (as Let's Encrypt probably does) rather than just the first one?

Reply:
for sure, totally agree. i wanted to get a simple implementation up and working and in front of you first. i'll make the changes. looks pretty simple, actually.

Reply:
well, it turns out that DnsClient.NET doesn't support this use case. i opened an [issue](https://github.com/MichaCo/DnsClient.NET/issues/43) inquiring why. i'm still going to do some minor code clean up and submit a PR. but i don't think we can address multiple name servers without jumping through a lot of hoops. also, while on that topic, DnsClient.NET rotates through registered name servers on subsequent calls so i don't even think it really adds value even if the client provided a list of all name servers for a given SOA. see [here](https://github.com/MichaCo/DnsClient.NET/issues/6) and [here](https://github.com/MichaCo/DnsClient.NET/issues/18) for more details.

Reply:
> i'm still going to do some minor code clean up and submit a PR.

addressed in https://github.com/PKISharp/win-acme/pull/1054/commits/0d89b527c7fdd49cad99962245faa00f4211bf1b

Reply:
Too bad that not all of the SOA records are returned. So I think we can forget about this idea for now, perhaps until the upstream library resolves your issue.